docs/user-guide/projects.md

```diff
@@ -54,6 +54,75 @@ argocd proj add-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 argocd proj remove-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 ```
 
+We can also do negations of destinations (i.e. install anywhere _apart from_).
+
+```bash
+argocd proj add-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+argocd proj remove-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+```
+
+Declaratively we could do something like this:
+
+```yaml
+spec:
+  destinations:
+  # Do not allow any app to be installed in `kube-system`  
+  - namespace: '!kube-system'
+    server: '*'
+  # Or any cluster that has a URL of `team1-*`   
+  - namespace: '*'
+    server: '!https://team1-*'
+    # Any other namespace or server is fine though.
+  - namespace: '*'
+    server: '*'
+```
+
+Doing this is order-dependent though, and this could also lead to unexpected behavior. Take this example:
+
+```yaml
+spec:
+  destinations:
+  # Allow any app to be installed in `kube-system`  
+  - namespace: 'kube-system'
+    server: '*'
+  # Or not?
+  - namespace: '!kube-system'
+    server: '*'
+```
+
+This would mean that any app can be allowed to be installed into `kube-system`, which is _probably_ not what you wanted.
+To get around this we have a field on the project spec called `allDestinationsMustMatch`, which means that all the 
+`destination` rules must be `true` for a given destination. 
+
+```yaml
+spec:
+  # With this field in place, nothing can be installed in `kube-system`.
+  allDestinationsMustMatch: true
+  destinations:
+  - namespace: 'kube-system'
+    server: '*'
+  - namespace: '!kube-system'
+    server: '*'
+```
+
+The first example above can be rewritten as this with `allDestinationsMustMatch`.
```

Comment:
I wouldn't re-order the elements here. It might reinforce the false idea that order is significant. I'd just copy the example again and emphasize that the behavior is now what we expected, thanks to the `allDestinationsMustMatch` line.